src/core/aws-utils.adb

```diff
@@ -70,6 +70,12 @@ package body AWS.Utils is
 
    Random_Generator : Integer_Random.Generator;
 
+   protected Shared_Random is
+      function Generate return Random_Integer;
+   end Shared_Random;
+
+   Random_Task_Safe : Boolean := False;
```

Comment:
Should be safe by default. User should not care about this except if performance is an issue.

Reply:
this flag removed. We are protected always.